src/requests/utils.ts

```diff
@@ -14,15 +14,7 @@ const normalizeHeaderValue = (value: string) => value.trim()
 export const normalizeHeaders = (headers: Record<string, string>) =>
   map(headers, ([key, value]) => [normalizeHeaderKey(key), normalizeHeaderValue(value)])
 
-export const pickHeaders = (headers?: Record<string, string>, keys?: string[]) => {
-  if (!headers) {
-    return {}
-  }
-
-  if (!keys) {
-    return headers
-  }
-
+export const pickHeaders = (headers: Record<string, string>, keys: string[]) => {
```

Comment:
Headers are generally speaking not mandatory in our model. This should stay optional for consistency

Reply:
These `if` statements are unused, even if the user passes no headers, because we already validate it beforehand and pass an empty object here